replication-api-impl/src/main/java/com/connexta/replication/api/impl/data/ReplicationItemManagerImpl.java

```diff
@@ -61,17 +67,33 @@ public ReplicationItemManagerImpl(ItemRepository itemRepository, SolrTemplate so
 
   @Override
   public ReplicationItem get(String id) {
-    return itemRepository
-        .findById(id)
-        .map(ReplicationItemImpl::new)
-        .orElseThrow(NotFoundException::new);
+    try {
+      return itemRepository
+          .findById(id)
+          .map(ReplicationItemImpl::new)
+          .orElseThrow(NotFoundException::new);
+    } catch (NonTransientDataAccessException e) {
+      throw new NonTransientReplicationPersistenceException(e);
+    } catch (TransientDataAccessException e) {
+      throw new TransientReplicationPersistenceException(e);
+    } catch (RecoverableDataAccessException e) {
+      throw new RecoverableReplicationPersistenceException(e);
+    }
   }
 
   @Override
   public Stream<ReplicationItem> objects() {
-    return StreamSupport.stream(itemRepository.findAll().spliterator(), false)
-        .map(ReplicationItemImpl::new)
-        .map(ReplicationItem.class::cast);
+    try {
+      return StreamSupport.stream(itemRepository.findAll().spliterator(), false)
+          .map(ReplicationItemImpl::new)
+          .map(ReplicationItem.class::cast);
+    } catch (NonTransientDataAccessException e) {
```

Comment:
❓ What was the purpose of wrapping these exceptions and throwing? Is a client expected to catch any of these that allow retry or recovery?

Reply:
They all inherit from a base one so if a client doesn't care about the specifics then they can just catch the base one. However, from documentation in Solr, each of these categories can be handled differently.